Sources/WalletConnectPairing/Services/Common/Delete/PairingDeleteService.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+public final class PairingDeleteService {
+    private let deleteRequester: DeleteRequester
+    private let deleteRequestSubscriber: DeleteRequestSubscriber
+    private let deleteResponseSubscriber: DeleteResponseSubscriber
+    
+    public var onDeleteRequestSubscriberResponse: ((String) -> Void)? {
+        get {
+            return deleteRequestSubscriber.onResponse
+        }
+        set {
+            deleteRequestSubscriber.onResponse = newValue
+        }
+    }
+    
+    public var onDeleteResponseSubscriberResponse: ((String) -> Void)? {
+        get {
+            return deleteResponseSubscriber.onResponse
+        }
+        set {
+            deleteResponseSubscriber.onResponse = newValue
+        }
+    }
+    
+    public init(
+        networkingInteractor: NetworkInteracting,
+        kms: KeyManagementServiceProtocol,
+        pairingStorage: WCPairingStorage,
+        logger: ConsoleLogging
+    ) {
+        let protocolMethod = PairingProtocolMethod.delete
```

Comment:
Any reasons to build dependencies inside init method? We normally doing it in client's factory